pkg/syncer/metadatasyncer.go

```diff
@@ -2640,3 +2655,139 @@ func initResizeReconciler(ctx context.Context, tkgClient clientset.Interface,
+		return
+	}
+	storageQuotaClient, err := k8s.NewClientForGroup(ctx, config, cnsoperatorv1alpha1.GroupName)
+	if err != nil {
+		log.Errorf("policyQuotaCRAdded: Failed to create CnsOperator client. Err: %+v", err)
+		return
+	}
+	storageClassList, err := k8sClient.StorageV1().StorageClasses().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		log.Errorf("policyQuotaCRAdded: Failed to list storageclasses. Err: %+v", err)
+		return
+	}
+	// For each storage class associated with storage policy id of StoragePolicyQuota CR,
+	// check if StoragePolicyUsage CR with resource type PVC exists.
+	// If not, create one with all parameters specified.
+	for _, sc := range storageClassList.Items {
+		if sc.Parameters[scParamStoragePolicyID] == storagePolicyId {
+			policyUsageList := &storagepolicyusagev1alpha1.StoragePolicyUsageList{}
+			err := storageQuotaClient.List(ctx, policyUsageList, &client.ListOptions{
+				Namespace: namespace,
+			})
+			if err != nil {
+				log.Errorf("policyQuotaCRAdded: Failed to list %v CRs associated in namespace %v. Err: %+v",
+					cnsoperatorv1alpha1.CnsStoragePolicyUsageSingular, namespace, err)
+				return
+			}
+			foundPvcUsageInstance := false
+			for _, usage := range policyUsageList.Items {
+				if usage.Spec.StoragePolicyId == storagePolicyId &&
+					usage.Spec.StorageClassName == sc.Name {
+					if usage.Spec.ResourceKind == ResourceKindPVC {
+						foundPvcUsageInstance = true
+						break
+					}
+				}
+			}
+			if !foundPvcUsageInstance {
+				pvcQuotaUsageInstanceName := sc.Name + "-" + "pvc-usage"
+				_, err := createStoragePolicyUsageCR(ctx, storageQuotaClient, pvcQuotaUsageInstanceName, namespace,
+					storagePolicyId, sc.Name, ResourceKindPVC, ResourceAPIgroupPVC, QuotaExtensionServiceName)
+				if err != nil {
+					log.Errorf("policyQuotaCRAdded: Failed to create %v CR for %v kind. Err: %+v",
+						cnsoperatorv1alpha1.CnsStoragePolicyUsageSingular, ResourceKindPVC, err)
+					return
+				}
+				log.Infof("policyQuotaCRAdded: Created %v for policy %v storageclass %v resourceKind %v",
+					cnsoperatorv1alpha1.CnsStoragePolicyUsageSingular, storagePolicyId, sc.Name, ResourceKindPVC)
+			}
+		}
+	}
+}
+
+// startStoragePolicyQuotaCRInformer creates and starts an informer for StoragePolicyQuota custom resource.
+func startStoragePolicyQuotaCRInformer(ctx context.Context, cfg *restclient.Config) error {
+	log := logger.GetLogger(ctx)
+	// Create an informer for StoragePolicyQuota instances.
+	dynInformer, err := k8s.GetDynamicInformer(ctx, cnsoperatorv1alpha1.GroupName,
+		cnsoperatorv1alpha1.Version, cnsoperatorv1alpha1.CnsStoragePolicyQuotaPlural, metav1.NamespaceAll, cfg, true)
+	if err != nil {
+		return logger.LogNewErrorf(log, "failed to create dynamic informer for %s CR. Error: %+v",
+			cnsoperatorv1alpha1.CnsStoragePolicyQuotaSingular, err)
+	}
+	policyQuotaInformer := dynInformer.Informer()
+	_, err = policyQuotaInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
```

Comment:
if for some reason syncer container is switching leadership and we miss this event, when new syncer container starts, will this event replayed and required usage CR instances will be created?

Reply:
To handle this, we need to look into informer queue and how to replay events that are missed or unsuccessful